### PR TITLE
PGvector service

### PR DESCRIPTION
**Pull request by ganesanintel**
## Description

The summary of the proposed changes as long as the relevant motivation and context.

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: comps/vectorstores/langchain/pgvector/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+```bash
+export POSTGRES_USER=testuser
+export POSTGRES_PASSWORD=testpwd
+export POSTGRES_DB=vectordb
+```
+
+## 3. Run Pgvector service
+
+```bash
+docker run --name some-postgres -e POSTGRES_USER=${POSTGRES_USER} -e POSTGRES_HOST_AUTH_METHOD=trust -e POSTGRES_DB=${POSTGRES_DB} -e POSTGRES_PASSWORD=${POSTGRES_PASSWORD} -d -v ./init.sql:/docker-entrypoint-initdb.d/init.sql pgvector/pgvector:0.7.0-pg16
```
Comment (amberjain1) [inline comps/vectorstores/langchain/pgvector/README.md:20]: Lets use a more meaningful name. Something like vectorstore-postgres?

Comment (ganesanintel) [reply comps/vectorstores/langchain/pgvector/README.md]: Done

**Review by amberjain1 — APPROVED** [review]

**Review by lvliang-intel — APPROVED** [review]

**Review by hshen14 — APPROVED** [review]

**Comment by chensuyue** [comment]
Please contribute an e2e test for this microservice, like this one: `https://github.com/opea-project/GenAIComps/blob/main/tests/test_reranks.sh`, please name it as `test_vectorstores.sh`.

File: comps/vectorstores/langchain/pgvector/__init__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2024 Intel Corporation
```
Comment (chensuyue) [inline comps/vectorstores/langchain/pgvector/__init__.py:1]: This license header is out of date, please leave it empty (the bot will add it automatically) or add the new one,   
```
Copyright (C) 2024 Intel Corporation   
SPDX-License-Identifier: Apache-2.0   
```

File: comps/vectorstores/langchain/pgvector/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Start PGVector server
+
+## 1. Download Pgvector image
+
+```bash
+docker pull pgvector/pgvector:0.7.0-pg16
```
Comment (chensuyue) [inline comps/vectorstores/langchain/pgvector/README.md:6]: This should be part of the BoM list, please fill in.

Comment (ganesanintel) [reply comps/vectorstores/langchain/pgvector/README.md:6]: Done

File: comps/vectorstores/langchain/pgvector/__init__.py (hunk 2 above)
Comment (ganesanintel) [reply comps/vectorstores/langchain/pgvector/__init__.py:1]: Addressed in https://github.com/opea-project/GenAIComps/pull/201

**Comment by ganesanintel** [comment]
> Please contribute an e2e test for this microservice, like this one: `https://github.com/opea-project/GenAIComps/blob/main/tests/test_reranks.sh`, please name it as `test_vectorstores.sh`.

Done. Added a testcase in https://github.com/opea-project/GenAIComps/pull/201.

**Pull request merged**